tests/unit_tests/snapshot_test.py

```diff
@@ -0,0 +1,387 @@
+"""Unit tests for the Snapshot class."""
+
+from diffkemp.snapshot import Snapshot
+from diffkemp.llvm_ir.kernel_module import LlvmKernelModule
+from diffkemp.llvm_ir.kernel_source import KernelSource
+import datetime
+import yaml
+
+
+def test_snapshot_initialization():
```

Comment:
Replace this test and the following two test by test of the two constructors (if you implement them).